pkg/topology/full/full.go

```diff
@@ -20,81 +21,107 @@ func init() {
 	rand.Seed(time.Now().UnixNano())
 }
 
-var _ topology.Driver = (*driver)(nil)
+var _ topology.Driver = (*Driver)(nil)
 
-// driver drives the connectivity between nodes. It is a basic implementation of a connectivity driver.
+// Driver drives the connectivity between nodes. It is a basic implementation of a connectivity Driver.
 // that enabled full connectivity in the sense that:
-// - Every peer which is added to the driver gets broadcasted to every other peer regardless of its address.
+// - Every peer which is added to the Driver gets broadcasted to every other peer regardless of its address.
 // - A random peer is picked when asking for a peer to retrieve an arbitrary chunk (Peerer interface).
-type driver struct {
-	mtx         sync.Mutex
-	connected   map[string]swarm.Address
-	discovery   discovery.Driver
-	addressBook addressbook.Getter
+type Driver struct {
+	discovery     discovery.Driver
+	addressBook   addressbook.GetPutter
+	connecter     p2p.Service
+	receivedPeers map[string]bool
```

Comment:
Nice, I will switch to `struct{}` and add comment.  I was also thinking about the infinite grow, but I did not go into it because it is probably not going to be a production solution, but you are right, there should be a comment at least, and let's add expiration or cleanup if/when needed!